ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import { SvgAdd, SvgDelete } from "@itwin/itwinui-icons-react";
+import { Button, ButtonGroup, IconButton, Select, SelectOption } from "@itwin/itwinui-react";
+import { UiComponents } from "../UiComponents";
+import { FilterBuilderContext } from "./FilterBuilder";
+import { FilterBuilderRuleRenderer } from "./FilterBuilderRule";
+import { FilterBuilderRuleGroup, FilterBuilderRuleGroupItem, isFilterBuilderRuleGroup } from "./FilterBuilderState";
+import { FilterRuleGroupOperator } from "./Operators";
+
+/** @alpha */
+export interface FilterBuilderRuleGroupRendererProps {
+  path: string[];
+  group: FilterBuilderRuleGroup;
+}
+
+/** @alpha */
+export function FilterBuilderRuleGroupRenderer(props: FilterBuilderRuleGroupRendererProps) {
+  const {path, group} = props;
+  const {dispatch} = React.useContext(FilterBuilderContext);
+
+  const addRule = () => dispatch({type: "ADD_ITEM", path, itemType: "RULE"});
+  const addRuleGroup = () => dispatch({type: "ADD_ITEM", path, itemType: "RULE_GROUP"});
+  const removeGroup = () => dispatch({type: "REMOVE_ITEM", path});
+
+  const onOperatorChange = React.useCallback((operator) => {
```

Comment:
It does not look like `Button` component is memoized to avoid rerendering on property changes and does not look like it has any effects that depend on `onClick` callback so I do not see the need to have a steady reference for `onClick` handlers.